src/parsing/Parse_rule.ml

```diff
@@ -1322,6 +1322,9 @@ let parse_taint_source ~(is_old : bool) env (key : key) (value : G.expr) :
       take_opt dict env parse_bool "by-side-effect"
       |> Option.value ~default:false
     in
+    let source_control =
+      take_opt dict env parse_bool "control" |> Option.value ~default:false
```

Comment:
should this be an options: of the rule instead?

Reply:
It could be that way too, but having it as an option in the source gives us more flexibility. A sink can now depend on both data and control sources.